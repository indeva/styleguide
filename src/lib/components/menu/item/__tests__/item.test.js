import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Item from '..';

describe('Item', () => {
  describe('without links', () => {
    it('without url should render properly', () => {
      const component = renderer.create(<Item title="Test" />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with url should render properly', () => {
      const component = renderer.create(<Item title="Test" url="#" />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with url and target should render properly', () => {
      const component = renderer.create(
        <Item title="Test" url="#" target="_blank" />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with active truthly should render properly', () => {
      const component = renderer.create(
        <Item title="Test" active />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with active truthly should render properly', () => {
      const component = renderer.create(
        <Item title="Test" active />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with icon should render with icon', () => {
      const component = renderer.create(
        <Item title="Test" icon={['fal', 'igloo']} />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('with links', () => {
    const defaultProps = {
      title: 'Test',
      active: false,
      links: [
        { title: 'Test' },
        { title: 'Test2', active: true },
        { title: 'Test3', url: '#' },
        { title: 'Test4', url: '#', target: '_blank' },
      ],
    };

    it('should render properly', () => {
      const component = renderer.create(
        <Item {...defaultProps} />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    describe('when item is active', () => {
      it('should render properly', () => {
        const component = renderer.create(
          <Item {...defaultProps} active />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render dropdown with active modifier', () => {
        const wrapper = mount(
          <Item {...defaultProps} active />,
        );
        const dropdown = wrapper.find('.menu__dropdown');
        expect(dropdown.hasClass('menu__dropdown--active')).toBeTruthy();
      });
    });

    describe('behavior tests', () => {
      describe('expanded inicial state', () => {
        it('starts expanded as true when active', () => {
          const wrapper = mount(
            <Item
              title="test"
              links={[
                { title: 'test' },
              ]}
              active
            />,
          );

          expect(wrapper.state('expanded')).toBeTruthy();
        });

        it('starts expanded as false when active', () => {
          const wrapper = mount(
            <Item
              title="test"
              links={[
                { title: 'test' },
              ]}
            />,
          );

          expect(wrapper.state('expanded')).toBeFalsy();
        });
      });

      describe('when the parent links has been clicked', () => {
        it('calls the toggle expanded state and calls callback functioon', () => {
          const onClickCallback = jest.fn();
          const wrapper = mount(
            <Item
              title="test"
              links={[
                { title: 'test' },
              ]}
              active
              onClick={onClickCallback}
            />,
          );

          expect(wrapper.state('expanded')).toBeTruthy();

          wrapper.find('.menu__link--active').last().simulate('click');

          expect(wrapper.state('expanded')).toBeFalsy();
          expect(onClickCallback).toHaveBeenCalledTimes(1);
        });
      });
    });
  });
});
