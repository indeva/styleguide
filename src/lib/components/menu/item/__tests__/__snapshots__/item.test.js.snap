// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item with links should render properly 1`] = `
<li
  className="menu__item"
>
  <span
    className="menu__link"
    onClick={[Function]}
    role="presentation"
  >
    <span
      className="menu__item-text"
    >
      Test
    </span>
  </span>
  <ul
    className="menu__dropdown"
  >
    <li
      className="menu__item menu__item--child"
    >
      <span
        className="menu__link menu__link--child"
        onClick={null}
        role="presentation"
      >
        Test
      </span>
    </li>
    <li
      className="menu__item menu__item--child"
    >
      <span
        className="menu__link menu__link--child menu__link--active"
        onClick={null}
        role="presentation"
      >
        Test2
      </span>
    </li>
    <li
      className="menu__item menu__item--child"
    >
      <a
        className="menu__link menu__link--child"
        href="#"
        target="_self"
      >
        Test3
      </a>
    </li>
    <li
      className="menu__item menu__item--child"
    >
      <a
        className="menu__link menu__link--child"
        href="#"
        target="_blank"
      >
        Test4
      </a>
    </li>
  </ul>
</li>
`;

exports[`Item with links when item is active should render properly 1`] = `
<li
  className="menu__item"
>
  <span
    className="menu__link menu__link--active"
    onClick={[Function]}
    role="presentation"
  >
    <span
      className="menu__item-text"
    >
      Test
    </span>
  </span>
  <ul
    className="menu__dropdown menu__dropdown--active"
  >
    <li
      className="menu__item menu__item--child"
    >
      <span
        className="menu__link menu__link--child"
        onClick={null}
        role="presentation"
      >
        Test
      </span>
    </li>
    <li
      className="menu__item menu__item--child"
    >
      <span
        className="menu__link menu__link--child menu__link--active"
        onClick={null}
        role="presentation"
      >
        Test2
      </span>
    </li>
    <li
      className="menu__item menu__item--child"
    >
      <a
        className="menu__link menu__link--child"
        href="#"
        target="_self"
      >
        Test3
      </a>
    </li>
    <li
      className="menu__item menu__item--child"
    >
      <a
        className="menu__link menu__link--child"
        href="#"
        target="_blank"
      >
        Test4
      </a>
    </li>
  </ul>
</li>
`;

exports[`Item without links with active truthly should render properly 1`] = `
<li
  className="menu__item"
>
  <span
    className="menu__link menu__link--active"
    onClick={[Function]}
    role="presentation"
  >
    <span
      className="menu__item-text"
    >
      Test
    </span>
  </span>
</li>
`;

exports[`Item without links with active truthly should render properly 2`] = `
<li
  className="menu__item"
>
  <span
    className="menu__link menu__link--active"
    onClick={[Function]}
    role="presentation"
  >
    <span
      className="menu__item-text"
    >
      Test
    </span>
  </span>
</li>
`;

exports[`Item without links with icon should render with icon 1`] = `
<li
  className="menu__item"
>
  <span
    className="menu__link"
    onClick={[Function]}
    role="presentation"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-igloo fa-w-18 menu__icon"
      data-icon="igloo"
      data-prefix="fal"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M288 32C128.9 32 0 160.9 0 320v128c0 17.7 14.3 32 32 32h512c17.7 0 32-14.3 32-32V320c0-159.1-128.9-288-288-288zm199.5 128H352V72.4c54.2 14 101.5 45.2 135.5 87.6zM448 320h-49.8c-22.2-38.1-63-64-110.2-64s-88 25.9-110.2 64H128V192h320v128zM288 64c10.9 0 21.5.9 32 2.2V160H88.5c46.9-58.4 118.9-96 199.5-96zM66.6 192H96v128H32c0-46.7 12.7-90.3 34.6-128zM32 448v-96h132.5c-2.7 10.3-4.5 20.9-4.5 32v64H32zm160 0v-64c0-53 43-96 96-96s96 43 96 96v64H192zm352 0H416v-64c0-11.1-1.9-21.7-4.5-32H544v96zm-64-128V192h29.4c21.9 37.7 34.6 81.3 34.6 128h-64z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="menu__item-text"
    >
      Test
    </span>
  </span>
</li>
`;

exports[`Item without links with url and target should render properly 1`] = `
<li
  className="menu__item"
>
  <a
    className="menu__link"
    href="#"
    target="_blank"
  >
    <span
      className="menu__item-text"
    >
      Test
    </span>
  </a>
</li>
`;

exports[`Item without links with url should render properly 1`] = `
<li
  className="menu__item"
>
  <a
    className="menu__link"
    href="#"
    target="_self"
  >
    <span
      className="menu__item-text"
    >
      Test
    </span>
  </a>
</li>
`;

exports[`Item without links without url should render properly 1`] = `
<li
  className="menu__item"
>
  <span
    className="menu__link"
    onClick={[Function]}
    role="presentation"
  >
    <span
      className="menu__item-text"
    >
      Test
    </span>
  </span>
</li>
`;
